### PR TITLE
clean up code for diagnosis module

### PR DESCRIPTION
**Pull request by vinay-debug**
### Related issue
  - *Issue Id*
  - *Context*
  
### List of changes
  - *change a*
  - *change b*

### Database migration/schema changes (if any)
- *List your migrations here...*

### Self review checklist
  - [ ] Confirm that the changes made by you meet the resolution criteria and nothing else.
  - [ ] Ran all the tests locally to check that you have not introduced any new regression.
  - [ ] If there are any failing tests, you have already fixed them. If not 
  - [ ] Followed coding and styling guidelines.
  - [ ] Made sure the changes are is not overly complex, unreadable and unnecessarily smart. Simplicity wins!
  - [ ] Made sure the comments are not too verbose and not to cryptic.
  - [ ] Made sure the code metrics are not deteriorated.
  - [ ] Made sure you have updated (if necessary)-
    - [ ] Documentation
    - [ ] Translation
    - [ ] Tests/test data
    - [ ] Postman collection
    - [ ] Version
       
### Additional info (if any)
- *Add anything you feel worth mentioning...*

### screensshots (if any)

Original | Updated
:------------------------:|-------------------------------|
** Original screenshot ** | ** Updated screenshot **|

### CODE REVIEW

File: src/api/controllers/clinical/diagnosis.controller.ts [hunk 1]
```diff
@@ -71,42 +58,49 @@ export class DiagnosisController {
...
-                throw new ApiError(404, 'Medical practitioner not found.');
+            ResponseHandler.success(request, response, 'Diagnosis retrieved successfully!', 200, {
+                Diagnosis : diagnosis,
+            });
+        } catch (error) {
+            ResponseHandler.handleError(request, response, error);
+        }
+    };
+
+    getByEvent = async (request: express.Request, response: express.Response): Promise<void> => {
```
Comment (kiran-rean) [inline src/api/controllers/clinical/diagnosis.controller.ts:78]: validation status is not an event. It is just a state of the diagnosis.
Event is something like doctor visit,

Comment (kiran-rean) [reply src/api/controllers/clinical/diagnosis.controller.ts:78]: You can remove getByEvent from Diagnosis pipeline.

**Review by kiran-rean — CHANGES_REQUESTED** [review]
Please have a look at the comments and make the changes suggested.

File: src/api/controllers/clinical/diagnosis.controller.ts (hunk 1 above)
Comment (vinay-debug) [reply src/api/controllers/clinical/diagnosis.controller.ts:78]: I made this by thinking. one can get the diagnosis of how many persons confirmed positive or negative for the disease.
if it is not relevant then, I will remove.

Comment (vinay-debug) [reply src/api/controllers/clinical/diagnosis.controller.ts]: I have removed the getByEvent and pushed the codes to this branch.

**Pull request merged**